pyrene/src/components/DateTimePicker/RangeSelector/DetectClickOutside.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  useEffect,
+  useRef,
+  ReactElement,
+} from 'react';
+
+export interface DetectClickOutsideProps{
+  children?: ReactElement,
+  listen?: boolean,
+  ignore?: Node,
+  onClick?: () => void,
+  onClickOutside?: () => void,
+}
+
+const DetectClickOutside: React.FC<DetectClickOutsideProps> = ({
+  listen,
+  onClickOutside,
+  ignore,
+  ...props
+}: DetectClickOutsideProps) => {
+
+  const container = useRef<HTMLDivElement>(null);
+
+  const handleEvent = (e: Event) => {
+    if (container.current?.contains(e.target as Node)) return;
+
+    // This ignore prop is used mostly for the buttons/links that toggle menus and drawers
+    if (ignore && ignore.contains && ignore.contains(e.target as Node)) {
+      return;
+    }
+
+    onClickOutside?.();
+  };
+
+
+  const onKeyDn = (e: KeyboardEvent) => {
+    // If the user hits ESC, it's considered a click outside!
+    if (e.code.toLocaleLowerCase() === 'escape') onClickOutside?.();
+    handleEvent(e);
+  };
+
+  useEffect(() => {
+    if (listen && onClickOutside) {
+      // Add listeners
+      document.addEventListener('mousedown', handleEvent, false);
+      document.addEventListener('touchend', handleEvent, false);
+      document.addEventListener('keydown', onKeyDn);
+
+      return () => {
+        // Remove listeners
+        document.removeEventListener('mousedown', handleEvent, false);
+        document.removeEventListener('touchend', handleEvent, false);
+        document.removeEventListener('keydown', onKeyDn);
+      };
+    }
+    return () => {};
```

Comment:
This line of code is not necessary.

Reply:
If I remove this line, then I get linting error. I think this way is cleaner than disabling the linting rule but if for any reason it is better to remove it and disable the rule please let me know.

Reply:
The RangeSelector component is on hold.